src/os/CMakeLists.txt

```diff
@@ -21,30 +21,36 @@ set(libos_srcs
   filestore/WBThrottle.cc
   filestore/ZFSFileStoreBackend.cc
   memstore/MemStore.cc
+  kstore/kv.cc
   kstore/KStore.cc
   kstore/kstore_types.cc
-  bluestore/kv.cc
-  bluestore/Allocator.cc
-  bluestore/BitmapFreelistManager.cc
-  bluestore/BlockDevice.cc
-  bluestore/BlueFS.cc
-  bluestore/bluefs_types.cc
-  bluestore/BlueRocksEnv.cc
-  bluestore/BlueStore.cc
-  bluestore/bluestore_types.cc
-  bluestore/ExtentFreelistManager.cc
-  bluestore/FreelistManager.cc
-  bluestore/KernelDevice.cc
-  bluestore/StupidAllocator.cc
-  bluestore/BitMapAllocator.cc
-  bluestore/BitAllocator.cc
   fs/FS.cc
   ${libos_xfs_srcs})
 
+if(HAVE_LIBAIO)
+  list(APPEND libos_srcs
+    bluestore/kv.cc
+    bluestore/Allocator.cc
+    bluestore/BitmapFreelistManager.cc
+    bluestore/BlockDevice.cc
+    bluestore/BlueFS.cc
+    bluestore/bluefs_types.cc
+    bluestore/BlueRocksEnv.cc
+    bluestore/BlueStore.cc
+    bluestore/bluestore_types.cc
+    bluestore/ExtentFreelistManager.cc
+    bluestore/FreelistManager.cc
+    bluestore/KernelDevice.cc
+    bluestore/StupidAllocator.cc
+    bluestore/BitMapAllocator.cc
+    bluestore/BitAllocator.cc
+  )
+endif(HAVE_LIBAIO)
+
 if(WITH_FUSE)
   list(APPEND libos_srcs
     FuseStore.cc)
-endif()
+endif(WITH_FUSE)
```

Comment:
@tchaikov @liewegas
Right, this is the deal:
```
> diff -s src/os/bluestore/kv.cc src/os/kstore/kv.cc
Files src/os/bluestore/kv.cc and src/os/kstore/kv.cc are identical
```

So we should ditch one of the 2. And since there is nothing special bluestore with it, I suggest we delete 
`src/os/bluestore/kv.cc`
Unless Sage has a special meaning with kv.cc being copied to bluestore.